src/Workspaces/Core/Portable/FindSymbols/ReferenceLocationExtensions.cs

```diff
@@ -52,6 +52,10 @@ private static void AddSymbols(
     {
         foreach (var reference in references)
         {
+            // Filter out name-only references (e.g. nameof expressions, typeof expressions)
+            // This fixes the most common Call Hierarchy false positives
+            if (reference.SymbolUsageInfo.ValueUsageInfoOpt?.IsNameOnly() == true)
+                continue;
```

Comment:
Prefer an empty line following if statements. This acts as a visual break in the code. This bit of code up here does this and now we are going to do something else down here.
```suggestion
                continue;

```